src/components/MapLegend.tsx

```diff
@@ -0,0 +1,125 @@
+import { ExpressionName, FillPaint } from "mapbox-gl";
+import React, { ReactElement } from "react";
+import { renderToStaticMarkup } from "react-dom/server";
+import { ControlPosition, IControl, MapboxMap, useControl } from "react-map-gl";
+
+import "../app/mapLegend.css";
+
+interface LegendOptions {
+  position: ControlPosition;
+  source: string;
+  paint: FillPaint;
+  metadata: { [key: string]: string };
+}
+
+class MapLegendControlClass implements IControl {
+  private _map: MapboxMap | undefined;
+  private _container: HTMLElement;
+
+  constructor(options: LegendOptions) {
+    this._container = document.createElement("div");
+    this._container.classList.add("mapboxgl-ctrl", "mapboxgl-ctrl-legend");
+    // render legend as static markup so it can be rendered with map
+    this._container.innerHTML = renderToStaticMarkup(
+      <MapLegend {...options} />
+    );
+  }
+
+  // able to add event listeners here for interactivity with layer
+  onAdd = (map: MapboxMap) => {
+    this._map = map;
+    return this._container;
+  };
```

Comment:
now we should be able to add event listeners for the map, expand functionality and maybe add interactivity